hid_enabled.go

```diff
@@ -70,21 +107,117 @@ func Supported() bool {
 //  - If the vendor id is set to 0 then any vendor matches.
 //  - If the product id is set to 0 then any product matches.
 //  - If the vendor and product id are both 0, all HID devices are returned.
-func Enumerate(vendorID uint16, productID uint16) []DeviceInfo {
+func Enumerate(vendorID uint16, productID uint16) ([]DeviceInfo, error) {
 	enumerateLock.Lock()
 	defer enumerateLock.Unlock()
 
+	var infos []DeviceInfo
```

Comment:
Please put the libusb enumeration code into its own function. Enumerate is super long already.

Reply:
done